Automattic/wp-calypso
# Remove Set as Homepage from Pages context menu

This pull request removes the Set as Homepage context menu item for page cards in Calypso Pages.

There are two ways to set a front page and a posts page for a site: using Customizer, or wp-admin. Calypso Pages does not handle setting the posts page at all, and setting the homepage has several caveats that make it not so intuitive to use.

As we are working on improving the Calypso Pages section, this PR removes the homepage setting option to cause less confusion for our users.

This PR partially of fully reverts the following ones (there may be others affected):

- https://github.com/Automattic/wp-calypso/pull/9873
- https://github.com/Automattic/wp-calypso/pull/9092
- https://github.com/Automattic/wp-calypso/pull/9402
- https://github.com/Automattic/wp-calypso/pull/8863
- https://github.com/Automattic/wp-calypso/pull/8617
- https://github.com/Automattic/wp-calypso/pull/8721
- https://github.com/Automattic/wp-calypso/pull/8711

### Testing instructions

- Run the unit tests for the Redux code and make sure things pass:
    - `npm run test-client -- --grep "state sites"`
    - `npm run test-client -- --grep "state pages"`

- Manually set up the following 4 scenarios in Customizer or wp-admin and verify that the Pages section works as expected:
    - your front page is showing your most recent blog posts
    - your front page is a static page, there is no blog page set
    - your front page is a static page, and there is a blog page set
    - your front page is not set, but there is a blog page set
